kube_log_watcher/main.py

```diff
@@ -233,17 +234,22 @@ def get_new_containers_log_targets(
             kwargs['node_name'] = CLUSTER_NODE_NAME
             kwargs['pod_annotations'] = pod_annotations
 
-            if not all([kwargs['application_id'], kwargs['application_version']]):
-                if strict_labels:
-                    logger.warning(
-                        ('Labels "{}" and "{}" are required for container({}: {}) in pod({}) '
-                         '... Skipping!').format(APP_LABEL, VERSION_LABEL, container_name, container['id'], pod_name))
-                    continue
-                else:
-                    if not kwargs['application_id']:
-                        logger.warning('Cannot determine application_id for pod: {}. Falling back to pod name!'.format(
-                            pod_name))
-                        kwargs['application_id'] = kwargs['pod_name']
+            pod_strict_labels = {}
+
+            for label in strict_labels:
+                value = pod_labels.get(label)
+                if value:
+                    pod_strict_labels[label] = value
```

Comment:
Why do we need ``pod_strict_labels``? It is identical to ``pod_labels``.

```python
In [1]: set(['application', 'version']) - set(['application', 'a',  'c'])
Out[1]: {'version'}

In [2]: set(['application', 'version']) - set(['application', 'a',  'c', 'version'])
Out[2]: set()
```

Reply:
Yes, this is unneeded code.